File: server_python/database/abstract.py
import json

class AbstractDataObject(object):

    def __init__(self, filepath, logger):
        self._filepath = filepath
        self._logger = logger
        self.db = None
        self._initDb()

    def _initDb(self):
        self.db = self._readFile()

    def _readFile(self):
        with open(self._filepath, 'r') as f:
            return json.loads(f.read())

    def _writeFile(self):
        with open(self._filepath, 'w') as f:
            f.write(json.dumps(self.db, indent=2))

File: server2/src/database/abstract.js
"use strict";
exports.__esModule = true;
var fs = require("fs");
var AbstractDataObject = /** @class */ (function () {
    function AbstractDataObject(filepath, logger) {
        this._filepath = filepath;
        this._logger = logger;
        this._initDb();
    }
    AbstractDataObject.prototype._initDb = function () {
        this.db = this._readFile();
    };
    AbstractDataObject.prototype._readFile = function () {
        return JSON.parse(fs.readFileSync(this._filepath, { encoding: 'utf-8' }));
    };
    AbstractDataObject.prototype._writeFile = function () {
        fs.writeFileSync(this._filepath, JSON.stringify(this.db, null, 2));
    };
    return AbstractDataObject;
}());
exports.AbstractDataObject = AbstractDataObject;
